### ISSUE TITLE
Preferences list

### ISSUE THREAD

**Issue by ppwadhwa** [napari/napari#2172]
List of all preferences to include in dialogue.  This is related to #992, but is breaking out just the list of preferences to include.  Other parts of the persistence discussion (JSON vs YAML for example) will be in separate issues.  

- [x] theme (superseding #943)
- [ ] monitor DPI (#820)
- [x] call order of plugins (after #937)
- [x] window position/geometry
- [x] magic-naming for layer (off by default in #1008)
- [ ] PySide2 vs PyQt5 preference (if both installed)
- [ ] default colormaps or color combinations (as discussed in #619)
- [x] highlight thickness for shapes / points layers
- [x] dask cache size maximum, and dask fusion settings (#1173 (comment)). https://github.com/napari/napari/pull/4436
- [ ] opt in for telemetry
- [ ] version updates (has user been asked)
- [x] font size for console
- [x] key/mouse bindings

notes: - Need to know if this the first launch, keep in mind plug-in specific settings

**Comment by goanpeca** [comment]
Ref: https://github.com/napari/napari/issues/992

**Comment by goanpeca** [comment]
For later:
> opt in for telemetry
Needs to add the actual telemetry

> version updates (has user been asked)
Auto update mechanism?

> key/mouse bindings
To be defined.

**Comment by goanpeca** [comment]
>PySide2 vs PyQt5 preference (if both installed)

It seems qtconsole does not like to be installed next to pyside2 ?

**Comment by ppwadhwa** [comment]
Hi y'all,
I am looking at this issue again to see if there is more that I can finish on the checklist above.  Gonzalo and I talked about it a bit and we can finish quite a few.  However, I was wondering about the following:

default colormaps or color combinations 
dask cache size maximum, and dask fusion settings 
magic-naming for layer

Are these still things we need/want?  

@sofroniewn @tlambert03 @jni @Carreau @goanpeca

**Comment by jni** [comment]
I've ticked magic naming as I don't think that needs a preference (others should feel free to dissent here, I don't feel strongly). Dask cache behaviour is still important, because even though it's much improved, users should still be able to deactivate it in cases where they want a computation to be re-run every time they slice or run a callback (as in the case of [napariboard](https://twitter.com/jnuneziglesias/status/1282869593137016832)). Colormaps I find a very low priority to be honest.

**Comment by Czaki** [comment]
PySide vs PyQt5 is really hard to set because it needs to be determined by the first import of any module that contains qt import. 

But if there is the decision to implement it I think that this setting should be stored in a separate file (simplify parsing, reduce the number of imports required to parse). There should be also a fallback code if the import fails for any reason (shared settings between environments).  

Maybe these settings should be handled only if `napai` is launched as a standalone application.

**Comment by ppwadhwa** [comment]
@tlambert03 @jni @sofroniewn   
Based on this: https://github.com/napari/napari/pull/820#issuecomment-630494209, it looks like we may not the user to change the monitor DPI scaling.  Is that right?  Or does this DPI setting mean something different than changing `QApplication.setAttribute(Qt.AA_EnableHighDpiScaling)`?

**Comment by goanpeca** [comment]
@ppwadhwa we can allow the user to set a specific value for the scaling if needed :)

**Comment by ppwadhwa** [comment]
references for DPI scaling:
https://github.com/spyder-ide/spyder/blob/01d9195b1a33d02f42faf3c8e0047412966a24ce/spyder/plugins/application/confpage.py#L170

https://github.com/spyder-ide/spyder/blob/01d9195b1a33d02f42faf3c8e0047412966a24ce/spyder/plugins/application/confpage.py#L246

**Comment by kcpevey** [comment]
@ppwadhwa its been a while since this has been updated. Looks like a few more of the check boxes are completed now. Can you provide an update?

**Comment by ppwadhwa** [comment]
There is one PR that never got merged that will address one more bullet point.   I'm going to revive that today.  Otherwise, there are still things left to be done.  
I think opt in for telemetry may be coming.  @goanpeca ?

**Comment by goanpeca** [comment]
>I think opt in for telemetry may be coming

I don't think so (yet?) :)

**Comment by jni** [comment]
It's definitely coming! LOL But yes, we want to be really careful about it so it might be a while.

**Comment by psobolewskiPhD** [comment]
The font size Preference now also affects the console.

**Comment by psobolewskiPhD** [comment]
I kinda think we can close this?
- monitor DPI is handled automagically
- Is there really a need for swapping Qt backends?
- telemetry is a no
- updating, if we ever want it to be a thing, would probably be via the plugin-manager?
- not sure how colormaps would be intended to work in Preferences. I think the solution there is to use cmap (https://github.com/napari/napari/issues/7555)

If we do decide we want one of these extras, it may be best to have a dedicated issue?
cc: @jni
